ddtrace/llmobs/_integrations/anthropic.py

```diff
@@ -33,3 +45,93 @@ def _set_base_span_tags(
+        input_messages = self._extract_input_message(messages)
+
+        span.set_tag_str(SPAN_KIND, "llm")
+        span.set_tag_str(MODEL_NAME, span.get_tag("anthropic.request.model") or "")
+        span.set_tag_str(INPUT_MESSAGES, json.dumps(input_messages))
+        span.set_tag_str(METADATA, json.dumps(parameters))
+        span.set_tag_str(MODEL_PROVIDER, "anthropic")
+        if err or resp is None:
+            span.set_tag_str(OUTPUT_MESSAGES, json.dumps([{"content": ""}]))
+        else:
+            output_messages = self._extract_output_message(resp)
+            span.set_tag_str(OUTPUT_MESSAGES, json.dumps(output_messages))
+
+        span.set_tag_str(METRICS, json.dumps(_get_llmobs_metrics_tags(span)))
+
+    def _extract_input_message(self, messages):
+        """Extract input messages from the stored prompt.
+        Anthropic allows for messages and multiple texts in a message, which requires some special casing.
+        """
+        if not isinstance(messages, Iterable):
+            log.warning("Anthropic input must be a list of messages.")
+
+        input_messages = []
+        for message in messages:
+            if not isinstance(message, dict):
+                log.warning("Anthropic message input must be a list of message param dicts.")
+                continue
+
+            content = message.get("content", None)
+            role = message.get("role", None)
+
+            if role is None or content is None:
+                log.warning("Anthropic input message must have content and role.")
+
+            if isinstance(content, str):
+                input_messages.append({"content": content, "role": role})
+
+            elif isinstance(content, list):
+                for block in content:
+                    if block.get("type") == "text":
+                        input_messages.append({"content": block.get("text", ""), "role": role})
+                    elif block.get("type") == "image":
+                        # Store a placeholder for potentially enormous binary image data.
+                        input_messages.append({"content": "([IMAGE DETECTED])", "role": role})
+                    else:
+                        input_messages.append({"content": str(block), "role": role})
+
+        return input_messages
+
+    def _extract_output_message(self, response):
+        """Extract output messages from the stored response."""
+        output_messages = []
+        content = _get_attr(response, "content", None)
+        role = _get_attr(response, "role", "")
+
+        if isinstance(content, str):
+            return [{"content": self.trunc(content), "role": role}]
+
+        elif isinstance(content, list):
+            for completion in content:
+                text = _get_attr(completion, "text", None)
+                if isinstance(text, str):
+                    output_messages.append({"content": self.trunc(text), "role": role})
+        return output_messages
+
+
+def _get_llmobs_metrics_tags(span):
```

Comment:
Can we make this be an `AnthropicIntegration` class method? Just to keep consistency with the other LLM integration classes

Reply:
Bump on this